File: cpp/3300_min_element/min_element.cpp
#include <cassert>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  int minElement(vector<int>& nums) {
    int ans = 37;
    for (int num : nums) {
      int dig = 0;
      while (num) {
        dig += num % 10;
        num /= 10;
      }
      ans = min(ans, dig);
    }
    return ans;
  }
};

int main() {
  vector<tuple<vector<int>, int>> tests{
      {{10, 12, 13, 14}, 1},
      {{1, 2, 3, 4}, 1},
      {{999, 19, 199}, 10},
  };

  for (auto [nums, expect] : tests) {
    assert(Solution().minElement(nums) == expect);
  }
}

File: rust/3300_min_element/min_element.rs
struct Solution;

impl Solution {
    pub fn min_element(nums: Vec<i32>) -> i32 {
        let mut ans = 37;
        for num in nums {
            let mut dig = 0;
            let mut n = num;
            while n > 0 {
                dig += n % 10;
                n /= 10;
            }
            ans = ans.min(dig);
        }
        ans
    }
}

fn main() {
    let tests = vec![
        (vec![10, 12, 13, 14], 1),
        (vec![1, 2, 3, 4], 1),
        (vec![999, 19, 199], 10),
    ];

    for (nums, expected) in tests {
        assert_eq!(Solution::min_element(nums), expected);
    }
}
